sorc/build_all.sh

```diff
@@ -82,20 +64,94 @@ while getopts ":a:dfghj:kuvwy" option; do
       ;;
   esac
 done
-
 shift $((OPTIND-1))
 
+# If no build system was specified, build for gfs forecast-only
+if [[ $# -eq 0 ]]; then
+   selected_systems="gfs"
+else
+   selected_systems="$*"
+fi
+
+supported_systems=("gfs" "gefs" "sfs" "gsi" "gdas" "all")
+# shellcheck disable=SC2034
+gfs_builds="gfs gfs_utils ufs_utils upp ww3_gfs"
+# shellcheck disable=SC2034
+gefs_builds="gefs gfs_utils ufs_utils upp ww3_gefs"
+# shellcheck disable=SC2034
+sfs_builds="sfs gfs_utils ufs_utils upp ww3_gefs"
+# shellcheck disable=SC2034
+gsi_builds="gsi_enkf gsi_monitor gsi_utils"
+# shellcheck disable=SC2034
+gdas_builds="gdas gsi_monitor gsi_utils"
+# shellcheck disable=SC2034
+all_builds="gfs gfs_utils ufs_utils upp ww3_gfs gefs ww3_gefs gdas gsi_enkf gsi_monitor gsi_utils"
+
 logs_dir="${HOMEgfs}/sorc/logs"
 if [[ ! -d "${logs_dir}" ]]; then
   echo "Creating logs folder"
   mkdir -p "${logs_dir}" || exit 1
 fi
 
-# Check final exec folder exists
-if [[ ! -d "${HOMEgfs}/exec" ]]; then
-  echo "Creating ${HOMEgfs}/exec folder"
-  mkdir -p "${HOMEgfs}/exec"
-fi
+# Jobs per build ("min max")
+declare -A build_jobs build_opts build_scripts
+build_jobs=(
+    ["gfs"]=8 ["gefs"]=8 ["sfs"]=8 ["gdas"]=8 ["gsi_enkf"]=2 ["gfs_utils"]=1 ["ufs_utils"]=1
+    ["ww3_gfs"]=1 ["ww3_gefs"]=1 ["gsi_utils"]=1 ["gsi_monitor"]=1 ["gfs_utils"]=1 ["upp"]=1
+)
```

Comment:
Would love to pull this out of the yaml at some future point to eliminate duplication (unless we think we'll use fewer CPUs in interactive mode to avoid stressing the login nodes).

Reply:
Yeah, I think they will need to be separate.  I've overloaded the head nodes in the past (and had a finger-wagging from RDHPCS), while for the compute nodes we can hit them as hard as we want.  This is also why I am getting rid of the `-j` option.

Reply:
would love for this script to be able to read that yaml (or build_config) file to get that information.

Reply:
Maybe these could be set differently in the YAML depending on whether it's a compute-node or head-node build...